### PR TITLE
[EPIC] e2e tests MVP

### PR DESCRIPTION
**Pull request by kael89**
Includes already approved PRs:
* https://github.com/beyondessential/tupaia/pull/2132
* https://github.com/beyondessential/tupaia/pull/2178
* https://github.com/beyondessential/tupaia/pull/2161
* https://github.com/beyondessential/tupaia/pull/2197
* https://github.com/beyondessential/tupaia/pull/2200
* https://github.com/beyondessential/tupaia/pull/2216

### CODE REVIEW

**Comment by kael89** [comment]
**Manual steps required** after this is merged in `dev` (and `prod`):
- [ ] Add the following `e2e` environment variables in `web-frontend/.env`:
   * `CYPRESS_BASE_URL`
   * `CYPRESS_RECORD_KEY`
   *  `CYPRESS_SNAPSHOT_REPO_URL`
   *  `GITHUB_ACCESS_TOKEN`
- [ ] Install [Cypress prerequisites](https://docs.cypress.io/guides/continuous-integration/introduction.html#Dependencies) on the EC2 instance

**Review by edmofro — APPROVED** [review]

**Comment by kael89** [comment]
@edmofro Mind if I rebase this off `dev` locally and force push in the feature branch? That way we'll have a cleaner commit history (avoid the `merge dev` commits)

**Comment by kael89** [comment]
@edmofro I can't do it via the UI cause there is a trivial conflict in `validateBranchName.sh`

**Comment by edmofro** [comment]
Yes sure :-)

File: packages/web-frontend/cypress/scripts/testE2e/pushSnapshots.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { SNAPSHOTS } from '../../constants';
 
 export const pushSnapshots = async (snapshotRepo, branch, snapshots) => {
   const date = Date.now();
-  const updateBranch = `tupaiabot/snapshot-update/${branch}_${date}`;
+  const updateBranch = `beyondessential-bot/snapshot-update/${branch}_${date}`;
```
Comment (kael89) [inline packages/web-frontend/cypress/scripts/testE2e/pushSnapshots.js:12]: @edmofro I sneakily renamed the bots in this "minor fixes" commit while rebasing

**Comment by kael89** [comment]
@edmofro one of the [manual steps](https://github.com/beyondessential/tupaia/pull/2147#issuecomment-789302931) required for releasing this epic is installing some [Cypress prerequisites](https://docs.cypress.io/guides/continuous-integration/introduction.html#Dependencies) in the server. I've done so successfully in a few feature instances, are you fine with me installing those in production today?

This will help me run the tests before Friday's release

**Comment by edmofro** [comment]
Yeah go for it :+1:

**Pull request merged**